src/util/inkscape.js

```diff
@@ -0,0 +1,150 @@
+const fs = require('fs')
+const path = require('path')
+const childProcess = require('child_process')
+const parallel = require('run-parallel')
+const series = require('run-series')
+const uuid = require('uuid/v4')
+const os = require('os')
+const PNG = require('pngjs').PNG
+
+const PATH_TO_BUILD = path.join(os.tmpdir(), 'orca-build')
+try {
+  fs.mkdirSync(PATH_TO_BUILD)
+} catch (e) {}
+
+const fullyTransparentColorbarRegexp = /<rect class="cbbg"(\s+[-\w]+="[^"]+")*\s+style="[^"]*fill-opacity: 0;[^"]*"(\s+[-\w]+="[^"]+")*\/>/g
+const fullyTransparentRectRegexp = /<rect(\s+[-\w]+="[^"]+")*\s+style="[^"]*stroke-opacity: 0;[^"]*fill-opacity: 0;[^"]*"(\s+[-\w]+="[^"]+")*\/>/g
+const transparentFillRectRegexp = /<rect(\s+[-\w]+="[^"]+")*\s+style="[^"]*fill-opacity: 0;[^"]*"(\s+[-\w]+="[^"]+")*\/>/g
+const transparentStrokeRectRegexp = /<rect(\s+[-\w]+="[^"]+")*\s+style="[^"]*stroke-opacity: 0;[^"]*"(\s+[-\w]+="[^"]+")*\/>/g
+const imgRegexp = /<image\s+xmlns="http:\/\/www.w3.org\/2000\/svg"\s+xlink:href="data:image\/png;base64,([^"]*)"[^>]*>/
+const colorbarFillRegexp = /<rect class="cbfill"[^>]*\/>/
+const xlinkHrefDataImageRegexp = /xlink:href="data:image\/png;base64,([^"]*)"/
+
+/** Node wrapper for Inkscape
+ *
+ * $ apt-get install inkscape
+ * ... or on OS X:
+ * $ brew install inkscape
+ *
+ * See:
+ * - https://linux.die.net/man/1/inkscape
+ * - https://en.wikipedia.org/wiki/Inkscape
+ */
+class Inkscape {
+  constructor (pathToInkscape) {
+    this.cmdBase = pathToInkscape || 'inkscape'
+  }
+
+  /** Convert SVG to EMF
+   *
+   * @param {buffer} svg : svg data buffer
+   * @param {object} opts
+   *  - id {string}
+   * @param {function} cb
+   *  - err {null || error}
+   *  - result {buffer}
+   */
+  svg2emf (svg, opts, cb) {
+    const id = opts.id || uuid()
+    const inPath = path.join(PATH_TO_BUILD, id + '-svg')
+    const outPath = path.join(PATH_TO_BUILD, id + '-emf')
+    const cmd = `${this.cmdBase} --file ${inPath} --export-emf ${outPath}`
+
+    const destroyTmpFiles = (cb) => parallel([
+      (cb) => fs.unlink(inPath, cb),
+      (cb) => fs.unlink(outPath, cb)
+    ], cb)
+
+    svg = this.cleanSvg(svg)
+
+    series([
+      (cb) => fs.writeFile(inPath, svg, 'utf-8', cb),
+      (cb) => childProcess.exec(cmd, cb),
+      (cb) => fs.readFile(outPath, cb)
+    ], (err, result) => {
+      destroyTmpFiles(() => {
+        cb(err, result[2])
+      })
+    })
+  }
+
+  cleanSvg (svg) {
```

Comment:
Oh wow. That's some fancy regex work!

How you tried using something like [`xml2.js`](https://www.npmjs.com/package/xml2js) to loop over all those SVG nodes? It should make things easier to maintain.

Reply:
I was pondering whether to use regex vs parsing the XML. The latter is certainly less error-prone.

Now that I know what cleaning needs to be done, I will look into `xml2.js`!